File: backend-java/src/main/java/com/groceriesapp/repository/ShoppingListRepository.java
package com.groceriesapp.repository;

import com.groceriesapp.model.ShoppingList;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface ShoppingListRepository extends JpaRepository<ShoppingList, Long> {
    Optional<ShoppingList> findByHouseholdId(Long householdId);
}

File: backend/src/main/kotlin/com/groceriesapp/repository/ShoppingListRepository.kt
package com.groceriesapp.repository

import com.groceriesapp.model.ShoppingList
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface ShoppingListRepository : JpaRepository<ShoppingList, Long> {
    fun findByHouseholdId(householdId: Long): ShoppingList?
}
